_posts/2017-08-02-July.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title: "Dev Diary (July 2017)"
+author: "Corey Montella"
+tags: []
+---
+
+We've had quite a busy July! Let's get to it:
+
+## v0.4 milestones
+
+In July we started using v0.4 to build internal projects, in order to find bugs in the runtime and test the limits of its performance. If you've hit performance walls with the v0.3 runtime, you'll be happy to know the Rust runtime is several orders of magnitude (10x - 100x) faster. Beyond the runtime itself, we've started fleshing out the tooling:
+
+- Server - The sever works like it did in v0.2, hosting an Eve instance listening on a websocket. We also include a static file server that hosts the output of your program.
+- Browser Client - The browser client connects to a remote Eve server to which it streams records like key presses; and receives records like ticks of a timer. These changes are rendered to the browser by watchers, which extend Eve's functionality into a foreign language. So far we have interfaces for writing watchers in Rust and Typescript.
+
+- Persistance - This is one of the features that has been missing for a while and frequently requested. Persistence is implemented by writing to an append-only binary file at a user specified location. Right now this is a flag passed to the Eve executable, but this could be controlled from within Eve itself. Transactions are written to this log as they are executed in the runtime. On load, the log of transactions is replayed, which restores the runtime to the state it was at when it was terminated. There are some obvious limitations to this when we start talking about Eve in a distributed context, but this at least gets us a starting place.
```

Comment:
"missing since 0.2" Anybody who hasn't followed 0.3 wouldn't think it's been missing. Worth noting that 0.4's persistence story is a lot faster/better too. Instead of storing the entire log and running through it all to load, it stores only what conservatively is required to rebuild the state.